src/Functions/GregorianDate.h

```diff
@@ -329,18 +329,29 @@ namespace DB
     }
 
     template <typename YearT>
-    OrdinalDate<YearT>::OrdinalDate(is_integer auto mjd)
+    OrdinalDate<YearT>::OrdinalDate(is_integer auto modified_julian_day)
```

Comment:
```suggestion
    template <is_integer TDay>
    OrdinalDate<YearT>::OrdinalDate(TDay modified_julian_day)
```
so you don't have to use decltype everywhere